baseband/dada/base.py

```diff
@@ -250,21 +251,69 @@ class DADAStreamReader(DADAStreamBase, VLBIStreamReaderBase):
         second selects channels.  If the tuple is empty (default), all
         components are read.
     """
+
     def __init__(self, fh_raw, squeeze=True, subset=()):
         fh_raw = DADAFileReader(fh_raw)
         header0 = DADAHeader.fromfile(fh_raw)
         super(DADAStreamReader, self).__init__(fh_raw, header0,
                                                squeeze=squeeze, subset=subset)
+        # Store number of frames, for finding last header.
+        raw_offset = self.fh_raw.tell()
+        self._nframes, self._pframe_nbytes = divmod(self.fh_raw.seek(0, 2),
+                                                    self.header0.frame_nbytes)
+        if self._pframe_nbytes > 0:
```

Comment:
For DADA, this works, but for GUPPI I'm not so certain because there are no restrictions on the length of a header.  In practice they shouldn't change for a single observation, but I don't know for sure.  (Not that we need the two to have identical algorithms.)

Reply:
Sure, will do.